### PR TITLE
fix(gpg): Try to shutdown GPG agent before removing its home directory

### PR DESCRIPTION
**Pull request by m-horky**
* Card ID: CCT-405

### CODE REVIEW

File: src/insights_client/__init__.py [hunk 1]
```diff
@@ -95,6 +95,34 @@ def sorted_eggs(eggs):
...
+    """
+    # Shut down GPG's home agent
+    shutdown_process = subprocess.Popen(
+        ["/usr/bin/gpgconf", "--homedir", home, "--kill", "all"],
+        stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+        universal_newlines=True,
+    )
+    stdout, stderr = shutdown_process.communicate()
+    if shutdown_process.returncode != 0:
+        log("Could not clean GPG home, got return code {rc}".format(rc=shutdown_process.returncode))
```
Comment (ptoscano) [inline src/insights_client/__init__.py:113]: hm IMHO this message can be easily mislead for the error that happens when executing `shutil.rmtree()`; maybe I'd be more explicit:
```suggestion
        log("Could not kill the GPG agent using gpgconf, got return code {rc}".format(rc=shutdown_process.returncode))
```

File: src/insights_client/__init__.py [hunk 2]
```diff
@@ -95,6 +95,34 @@ def sorted_eggs(eggs):
...
+        if stdout:
+            log(stdout)
+        if stderr:
+            log(stderr)
+
+    # Delete the temporary directory
+    try:
+        shutil.rmtree(home)
+    except Exception as exc:
+        log("Could not remove GPG home, got {exc}".format(exc=exc))
```
Comment (ptoscano) [inline src/insights_client/__init__.py:123]: should we still do not ignore the failures of `shutil.rmtree()` for now, to check whether anything wrong still happens on cleanup?

**Review by ptoscano — COMMENTED** [review]
The approach seems OK.

File: src/insights_client/__init__.py (hunk 2 above)
Comment (m-horky) [reply src/insights_client/__init__.py:123]: I am a bit scared to do that, but it might be a good idea.

**Review by ptoscano — APPROVED** [review]
Looking at the CI failures since they were introduced, the
```
FileNotFoundError: [Errno 2] No such file or directory: 'S.gpg-agent.extra'
```
error happens quite often, so hopefully this should help. In any case, we still have lots of time to stress this more with other PRs, and fix it again if needed.

LGTM, thanks!

**Comment by ptoscano** [comment]
Ah, the failure of the `testing-farm:fedora-40-x86_64:unit/fedora` job seems an infra issue; hence merging it regardless.

**Pull request merged**